state_signals.py

```diff
@@ -183,29 +184,39 @@ def __init__(
     ) -> None:
         """
         init: Sets exporter object fields and generates unique publisher_id.
-        Allows for specification of redis host/port. Will continue to attempt
-        to connect to redis server for conn_timeout seconds (or forever if
-        conn_timeout is set to -1). Also allows runner hostname to be inputted
-        manually (otherwise will default to platform.node() value).
+        Allows for use of an existing Redis connection object or specification
+        of redis host/port. Will continue to attempt to connect to redis server
+        for conn_timeout seconds (or forever if conn_timeout is set to -1). Also
+        allows runner hostname to be inputted manually (otherwise will default
+        to platform.node() value).
         """
         self.logger = _create_logger("SignalExporter", process_name, log_level)
         self.subs = set()
         self.proc_name = process_name
         self.runner_host = runner_host
-        self.pub_id = process_name + "-" + str(uuid.uuid4())
-        self.redis = redis.Redis(host=redis_host, port=redis_port, db=0)
+        self.failed_conn_attempts = 0
+        if not existing_redis_conn:
+            self.redis = redis.Redis(host=redis_host, port=redis_port, db=0)
+            self.conn_type = "new"
+        else:
+            self.redis = existing_redis_conn
+            self.conn_type = "existing"
         success = False
         while conn_timeout != 0:
             try:
                 self.redis.ping()
                 success = True
                 break
             except redis.ConnectionError:
-                time.sleep(1)
+                self.failed_conn_attempts += 1
                 conn_timeout -= 1
-                self.redis = redis.Redis(host=redis_host, port=redis_port, db=0)
+                time.sleep(1)
         if not success:
+            self.logger.critical(
```

Comment:
I don't think you want to use `critical`, just debug, because the code is used as a dependent module which already raises an error.  The logging can just add noise.